pkg/controllers/ingressnodesavailable/ingress_nodes_available_controller.go

```diff
@@ -53,7 +56,7 @@ func NewIngressNodesAvailableController(
 		).
 		WithSync(controller.sync).
 		ResyncEvery(wait.Jitter(time.Minute, 1.0)).
-		ToController("IngressNodesAvailableController", eventRecorder)
+		ToController(instanceName, eventRecorder)
```

Comment:
I think that in library-go we are passing `c.controllerInstanceName` (applies to the other changes in this PR)

Reply:
i'm fine either way but we should be consistent.

Reply:
given that for some controller we cannot change the name then maybe we should use `instanceName` ?

Reply:
That's my understanding as well.

Reply:
Actually I take it back. @deads2k created this m****s and was not willing to take into account feedback given in https://github.com/openshift/library-go/pull/1794 :(.

The `instanceName` is the name of the operator, for example, `openshift-apiserver` or `openshift-apiserver-operator`, and the `controllerInstanceName` is the `instanceName` combined with the name of the controller.

Given this, it doesn't make sense to use the `instanceName` as the controller's name, as this would result in multiple controllers sharing the same name.

For the `IngressNodesAvailableController`, we can change its name because the previous name was not used to generate a condition. However, for controllers created with `WithSyncDegradedOnError`, it is not safe to change their name.

Does that make sense ?

Reply:
[NewCSIDriverNodeServiceController](https://github.com/openshift/library-go/pull/1794/files/a86ed6242ce1d886cd79fbb1785829208d4d4f35#diff-9e9f771b56ad2865cbcbaf9907510ed027520a166736511b6364e68b11da543e) from the library-go PR mentioned earlier use `WithSyncDegradedOnError`, thus its name could not be changed. 

Given how this controller is [created](https://github.com/search?q=org%3Aopenshift%20NewCSIDriverNodeServiceController&type=code), the `instanceName` will be`NFSDriverNodeServiceController` and the `controllerInstanceName` will be `NFSDriverNodeServiceController-CSIDriverNodeService`

Reply:
Thanks for the explanation, @p0lyn0mial.
I'm worried that some variables, like `instanceName`, have different meanings in different places. For example, in this operator it would be `oauth-apiserver`, but in [storage](https://github.com/openshift/library-go/pull/1794/files/a86ed6242ce1d886cd79fbb1785829208d4d4f35#diff-9e9f771b56ad2865cbcbaf9907510ed027520a166736511b6364e68b11da543e)  it would be `NFSDriverNodeServiceController`, as you pointed out.